### PR TITLE
Migrate CBs and Sems to kernel config ring buffer

### PR DESCRIPTION
**Pull request by pgkeller**
### Ticket
#4984 

### Problem description
Migration of sems and cb configs to ring buffer

### What's changed
Move semaphores and CB configs to ring buffer.
Had to hack disable code in cq_dispatch due to code issues
Had to hack around race in llks (this code sped things up exposing a race, added a delay to compensate)

### Checklist
- [ ] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: tt_metal/hw/inc/blackhole/dev_mem_map.h [hunk 1]
```diff
@@ -52,7 +52,7 @@
 #define MEM_BOOT_CODE_BASE 0
 #define MEM_L1_BARRIER 12
 #define MEM_MAILBOX_BASE 16
-#define MEM_MAILBOX_END (MEM_MAILBOX_BASE + 1344)
+#define MEM_MAILBOX_END (MEM_MAILBOX_BASE + 1348)
```
Comment (dmakoviichuk-tt) [inline tt_metal/hw/inc/blackhole/dev_mem_map.h:55]: how do we calculate this value? Could you add comment or make it depend on sum of other defines directly?

**Review by aliuTT — APPROVED** [review]

File: tt_metal/hostdevcommon/common_runtime_address_map.h [hunk 2]
```diff
@@ -55,29 +55,19 @@ static_assert (PROFILER_FULL_HOST_BUFFER_SIZE_PER_RISC > PROFILER_L1_BUFFER_SIZE
 // Size is presently based on the old size of the RTAs (large enough to hold 1 set)
 // plus some extra space freed up in the mem map
 constexpr static std::uint32_t L1_KERNEL_CONFIG_BASE = PROFILER_L1_END_ADDRESS;
-constexpr static std::uint32_t L1_KERNEL_CONFIG_SIZE = 4 * 1024 + 256;
+constexpr static std::uint32_t L1_KERNEL_CONFIG_SIZE = 4 * 1024 + 256 + 128 + 512;
```
Comment (abhullar-tt) [inline tt_metal/hostdevcommon/common_runtime_address_map.h:58]: Can you update the comment above to specify that cbs and semaphores were added and the constant calculations for the respective sizes

File: tt_metal/hw/firmware/src/trisc.cc [hunk 3]
```diff
@@ -107,11 +108,17 @@ int main(int argc, char *argv[]) {
...
         DeviceZoneScopedMainN("TRISC-FW");
 
+        uint32_t kernel_config_base = mailboxes->launch.kernel_config.kernel_config_base;
+
 #if !defined(UCK_CHLKC_MATH)
-        setup_cb_read_write_interfaces(0, mailboxes->launch.kernel_config.max_cb_index, cb_init_read, cb_init_write);
+        uint32_t tt_l1_ptr *cb_l1_base = (uint32_t tt_l1_ptr *)(kernel_config_base +
+            mailboxes->launch.kernel_config.cb_offset);
+        setup_cb_read_write_interfaces(cb_l1_base, 0, mailboxes->launch.kernel_config.max_cb_index, cb_init_read, cb_init_write, cb_init_write);
+#if defined(UCK_CHLKC_UNPACK)
```
Comment (abhullar-tt) [inline tt_metal/hw/firmware/src/trisc.cc:117]: Can you add a comment about why this delay was added with the issue link?

File: tt_metal/hw/inc/wormhole/eth_l1_address_map.h [hunk 4]
```diff
@@ -28,7 +28,7 @@ struct address_map {
   static constexpr std::int32_t DATA_BUFFER_SIZE_ETH = 4 * 1024;
   static constexpr std::int32_t DATA_BUFFER_SIZE_NOC = 16 * 1024;
   static constexpr std::int32_t DATA_BUFFER_SIZE = 24 * 1024;
-  static constexpr std::int32_t ERISC_L1_KERNEL_CONFIG_SIZE = 96 * 4;
+  static constexpr std::int32_t ERISC_L1_KERNEL_CONFIG_SIZE = 96 * 4 + 8 * 16;
```
Comment (abhullar-tt) [inline tt_metal/hw/inc/wormhole/eth_l1_address_map.h:31]: Same comment as above  for adding comments for constants

**Review by abhullar-tt — APPROVED** [review]

**Review by tt-aho — APPROVED** [review]
Would be good to run some model ci, ex perf/device perf, to double check perf/also ensure the race workaround works for those tests

File: tt_metal/hw/firmware/src/trisc.cc (hunk 3 above)
Comment (pgkeller) [reply tt_metal/hw/firmware/src/trisc.cc:117]: yes, will do

**Comment by pgkeller** [comment]
Closing.  Semaphore change ran into major issues w/ T3k tests, fixed in a separate branch, will pull those in isolation then rebase the rest of this work on that.

**Pull request closed without merging**